app/controllers/works_controller.rb

```diff
@@ -0,0 +1,81 @@
+class WorksController < ApplicationController
+
+  before_action :require_login, only: [:new, :create, :update, :edit, :destroy]
+
+  def homepage
+    @works = Work.all
+  end
+
+  def index
+    @works = Work.all
+  end
+
+  def show
+    @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      flash.now[:error] = "Hmm..we couldn't find a work with that id"
+      redirect_to root_path #for now.. flash?
+      return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save
+      flash[:success] = "You've successfully created a new work!"
+      redirect_to work_path(id: @work[:id])
+      return
+    else
+      flash.now[:error] = "Hmm..something went wrong, your work was not saved"
+      render :new, status: :bad_request
+      return
+    end
+  end
+
+  def edit
+    @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      flash.now[:error] = "Hmm..we couldn't find a work with that id"
+      redirect_to works_path
+      return
+    end
+  end
+
+  def update
+    @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      flash.now[:error] = "Hmm..we couldn't find a work with that id"
+      render :not_found
+      return
+    else
+      @work.update(work_params)
+      flash[:success] = "You've successfully edited this work!"
+      redirect_to work_path(id: @work[:id])
```

Comment:
What if the update fails?

You should have an if to check and show validation errors.